### ISSUE TITLE
[Enhancement] Suggested sounds

### ISSUE THREAD

**Issue by OreCruncher** [OreCruncher/BetterRain#56]
This entry is to collect the suggestions and ideas for new and/or replacement sounds.  My goal is to avoid cluttering the Issue list with these types of suggestions as well as give a common point for folks to discuss.
- [ ] [Number station](https://archive.org/details/ird059) type block sounds for juke boxes and noteblocks
- [ ] [Use NOAA "unkown sounds"](https://en.wikipedia.org/wiki/List_of_unexplained_sounds) as spot sounds in Ocean/Deep Ocean
- [ ] Need new sound for underwater in River biome
- [ ] Low resonant erie sound for for End Stone
- [x] Hissing from grass/flowers (snake, animal)
- [x] Crops/plants rustling (crops do insect sounds; leaf/grass rustle)
- [x] Forestry/Magic Bee hives buzz
- [x] Paper rustling for bookshelves (random/step)
- [x] Monster Egg Blocks w/scratching, crawling, and hissing sounds
- [x] Nether Bricks heavy breathing
- [x] Desert sounds (sand blowing, rattlesnake, etc.)
- [x] Alligators in Swamps/marsh
- [x] Primates in jungle, savanna?
- [x] Buffalo in plains
- [x] Condition token: foggy
- [x] Condition token: freezing (temp <= 0.15)
- [ ] Village sounds - have a Village token when player within range of a village

**Comment by Sunconure11** [comment]
I'll post my sounds then, even though you covered some of them:

Random sounds:

Creaking wood
Rocks tumbling
Rocks being kicked
Gravel and sand being kicked up
Low, resonant sound from end stone (think stuff like whale song or even reconstructed hadrosaur songs (https://www.youtube.com/watch?v=d-tRFuMdQkA), but, more... eerie)
Hissing in grass and flowers
Labored breathing from Nether Bricks (Maybe we can make the nether just as creepy as it is in MATMOS)
Crawling in silverfish-infested stones, plus hissing

Sounds that would always play:

Crops and plants rustling (also has a random chance of playing even if not stepped on to simulate wind)
Ice skidding (plays always if you run on ice)
Papers rustling (if you step on a bookshelf, also might play in abandoned buildings with bookshelves at random)
Mud being kicked up (only plays in non-oceanic biomes and if you are stepping on dirt while underwater, also plays randomly)

Other sounds:

Dogs barking in villages
Soulsand has a chance to scream if stepped on
Anguished cries and screams in the nether
Cicada songs in forests
Buffalo calls in plains
Primate calls in jungles
Rattlesnake rattling in deserts
Musk oxen calls in taigas and tundras
Crocodile bellowing in swamps

**Comment by O13SC3N3** [comment]
Might be better with a few default sounds, and add a config option to load a custom json to allow the end user to add custom sounds of choice and events. Keeping the mod light weight and very versatile. 

Maybe options of spot or ambient etc, sound file location in surround folder in a folder named sounds.  Delay, chance etc.

So if someone wanted to make the Sirens in Silent Hill go off when (rain) fog thickens in the nether (which mine stopped happening for some reason) which would be freaking cool.

In resource packs you have the options of ambient as follows.cave 1-13.  Weather rain 1-4 thunder 1-3 and much more at

**Comment by O13SC3N3** [comment]
My suggestion would be to keep the mod lightweight and versatile as possible, allowing the user to the choice to make it as immersive or as heavy at will. Keep with a few default sounds that can be disabled at will.

Currently the mod has a lot of great features, Auroras, Spot Sounds, variable Dust/Rain/Snow storms. And more.

I would like to see the mod not take away from the users ability to use a resource pack to add sounds as well to given events, like Caves 1-13, etc, etc at http://minecraft.gamepedia.com/Sounds.json 

But capitalize on adding to that ability to make it more immersive, like the falling of boulders, kicking rocks, spotted by a wolf, the waves crashing on a beach (soft on sand, hard on rocks). Dynamic Surroundings can be a very strong tool in the end user with some information and a community creating custom sounds for events, spots, background w.e wether it be utilized in events and variables with the mod, and or resource packs.

So my suggestion comes in 2 parts one for OreCruncher and one for the Community who is loving this mod as much as I am.

For Dynamic Surroundings I suggest the ability to allow the end user to make the events/spots/etc when it comes to the sounds. Either with a json file like we use with adding fog to dimensions/biomes etc.

Example be like; if possible (Play a Siren like in the game Silent Hill, when the fog thickens in the Nether) and folder setup like Minecraft/Config/Dursurround (+ folder "Sounds") with additional jsons to the biomes and dimension options for weather and etc.

name: "Siren",
    dimension: "1"; (0, 1, -1, ALL)
    biome: "ALL", (maybe even in the Overworld, you could have it set to specific biomes like deserts)
    event: precipitation, (rain/whatever/spot?)
    sound: 'siren.mp3", (.ogg as well)
           loop: "false", (or true)
           dealy: 0; "0 for None, 5 for 30 seconds? etc)
           variation: (Maybe list a few more sound files here like)
                 "siren2.mp3",
                 "siren3.mp3",

I know its not in true json format but basically what Im getting at with this is in the Nether when it "rains" the fog thickens that was implemented with the mod, a Sound file will play the Siren like the silent hill siren at https://www.youtube.com/watch?v=OwFqShcxaQ0, and only do it once with the loop set to false and no delay, and has a chance to playing 1 of the 3? Which would be good with 3 different types of bird sounds in forest, or crickets w.e

now if I wanted this to do the same but in the overworld id do the same setup but in dimension 0, and have the biome ids or "DESERT" set to go off with precipitation again (DUST STORM).

So in the same we could do with the wolves spotting us as well. Or even when we see diamonds and it could play the sound file https://www.youtube.com/watch?v=BmvLexamrmk.

Ultimatey given us the end users to control how we want our worlds to act. And would give great control to even Adventure Map and HQM pack makers more immersive control.

Now for the Users I would like to see everyone come together either through a forum and collaborate or find make suggestions or even create sound resource packs (Like Snos etc) for minecrafts main means. And come up with our own sound packs for the frog croaks, wolves, whatever you may decide its your choice... You want flowers to sing Good morning to you, ok... go for it.

But that would be my ideal choice, and I dont know crap about java. Just an idea. Nothing against your work OreCruncher or choice of sounds. Just thinking you created a great tool, and you can add so many features, either way I will be following this mod for a long time.

**Comment by O13SC3N3** [comment]
another feature which would be cool with the biomes options or block updates be potion effects.

like if I were to script Biomes O Plenty; biome "Wasteland" to have increased oregen through CoFH or other means with IC2 Uranium/BC Yellowrite or some kind of metal like Cinnabar (In real life toxic) to give off IC2 Radiation or Poison effects.

Making it so if you were to enter within the wasteland or near large deposits of ore you can catch radiation debuff with out the Radiation Suit from IC2. maybe even play off in like Thaumcraft's Tainted Biome a chance to increase a players warp. Obviously controlled by the end user.

**Comment by OreCruncher** [comment]
Providing sound content can be a full time job. :)  A challenge I am having is finding good sounds that aren't encumbered with licensing or cost.  If I get enough through Patreon I should be able to start purchasing decent sounds for use in the mod.

I started looking at sound based resource packs.  Right now an author could create a resource pack that contains only sounds and drop them into the resourcepack folder.  These sounds can override what Vanilla or any other mod has created, and even add new ones that can be access by `/playsound` (if you like command blocks) or any other mod that is aware of them.  What Dynamic Surrounding needs to do is look into those packs for block and biome configuration Json so those can be incorporated into operation automatically.

Something that you touched on were sounds attached to a dimension.  It is something I want to do - just trying to figure out a decent way of actually integrating it.  Another thing is adding an additional conditions when a player is within "range" of a Villager so village sounds can trigger.  You know, these could be supplied as player sounds..

RE: potion effects for blocks.  At the moment I am trying to avoid altering the game mechanics.  Not ruling it out 100% forever and always, just for the moment.  The main reason I am doing this is to mitigate impacts on servers (performance) or require a lot of ASM to accomplish (modifying class code on the fly).  The original code I had for block updates used ASM to hook into Minecraft processing.  Problem I ran into was that it wasn't flexible - I had to look at the code for each block to make sure it would work, and this would became a problem when it came to modded blocks.  It quickly became tedious.

Last thing I am looking at is some JavaScript integration.  The condition filtering I have using regex works to a large degree, but doesn't have the granularity that an author may want.  For example, with regex you can't do something like "player's Y > 50" or "biome temp <= 0.15".  If I do add JavaScript it will be in additional to rather than replacing the regex - it would only execute if it is specified for a sound.  (If anyone has feedback on this I would appreciate it.)

**Comment by O13SC3N3** [comment]
With the dimension thing ie my siren example is it possible to call it when the event of rain your fog increase, and make it play the sound file?

**Comment by OreCruncher** [comment]
It will not work exactly like you are wanting.  For the siren example you want it to play once and then stop.  For it to work the way you want I would have to have some sort of memory where I keep track that it played.  If you did loop you could specify a repeat delay where it would play every N ticks.  So if you want you can play the siren sound, delay another 2 minutes, and play again.  Of course, you could set a delay of 60 minutes... (EDIT2: I use the repeat delay for the tummy grumble when the players hunger bar is low.  The sound is a regular background sound that is set to play when the conditions are met.)

In terms of what I have currently, the dimension name is included in the condition string.  The general form of the regex string would be something like `(?i)(.*overworld.*raining.*)`.  When attached to a biome that sound could play when the player is in that biome, in the Overworld dimension, and it is raining.  So if the sound is attached to Wasteland it will play in wasteland, in the Overworld dimension, and when it is raining.

The `Player` is a special biome.  You can think of the player as always being in the`Player` biome.  So if a sound with the above regex is configured it has a chance of playing in the Overworld dimension, regardless of the player's real biome.  This means that siren will sound anywhere in Overworld when it is raining, even under ground.

EDIT:  Forgot - can't do anything with fog at the moment.  This is where the JavaScript would come in handy (i.e. `fogLevel > 0.004`).  I could add a condition token for `foggy` but I am not sure what the threshold would be for setting it.

**Comment by OreCruncher** [comment]
Adding "freezing" and "fog" tokens for *.0.2.4BETA.  Freezing is when the biome temp at the player location is < 0.15, and fog is set when the Dynamic Surroundings effective fog level is >= 0.01.

**Comment by Sunconure11** [comment]
I found some sites with free sound clips:

https://www.freesound.org/

http://soundbible.com/free-sound-effects-1.html

http://www.hongkiat.com/blog/55-great-websites-to-download-free-sound-effects/

http://www.freesfx.co.uk/

You could look through these.

You could also make some new ones via Audacity. http://www.audacityteam.org/download/

For animal calls, you could rip a few from youtube videos.

**Comment by OreCruncher** [comment]
Yep - been looking at those sites.  I use Audacity to clean up some of the clips I have found.

**Comment by O13SC3N3** [comment]
249 Mods... feels so uneven... quick somebody give me a mod lol

**Comment by Sunconure11** [comment]
Totemic

**Comment by O13SC3N3** [comment]
Cool mod, but doesn;t really fit in my pack :p Ill have to keep that one in mind for new pack designs. Thanks.

**Comment by O13SC3N3** [comment]
@OreCruncher did you add ambiance to the end? It seems my servers resource pack (features certain song elements during combat and bosses) stopped working with the ender dragon.

**Comment by OreCruncher** [comment]
Yes, I put in some stuff.  Essentially turned auroras on, and there is a single biome sound for The End/Sky biome.  In your json you can reset the sound effects when you apply the sound you want.

(Other thing is that depending on the quantity of sounds queued in the system some of the sounds will cut out.  There is a config option in the latest BETA where you can increase the number of channels (sounds) the sound engine can play.)

**Comment by O13SC3N3** [comment]
Well I noticed regular music continued to play, ill mess around with it. Just the "Boss" event didn't trigger. Ill let you know if Dynamic Surroundings is the cause or not.

**Comment by Sunconure11** [comment]
Would it be possible to add your sounds to growthcraft bee hives and apiaries, plus the ones from harvestcraft?

https://github.com/GrowthcraftCE/Growthcraft-1.7

**Comment by OreCruncher** [comment]
@Sunconure11 It would beeee possible to do that.  However, the apiary one I may skip.  The reason is that players place them in concentrations and that will create a lot of bee noise.  By default the sound engine only allows 28 simultaneous "normal" sounds playing.

**Comment by OreCruncher** [comment]
Next beta will have floor squeeks for Growthcraft bamboo blocks, and bee sounds for Harvest craft hives.  (Growthcraft hives already have a bee sound.)  I didn't do the bee houses for those mods just yet.  I may do that because I don't think players will create farms of them like Forestry.

**Comment by O13SC3N3** [comment]
Harvest craft hives, forestry already have Bee sounds.

**Comment by Sunconure11** [comment]
What about snoring and sleep-talking from sleeping players, and if witchery is installed, sleeping villagers?

Sleep-talking sounds should be entirely androgynous, or at the very least, cannot be associated as being either male or female, for consistency.

**Comment by O13SC3N3** [comment]
chances of seeing people sleep in a bed is slim. Most servers run mods or plugins like Morpheus to out vote players to change it to day rather quickly. and in a single player world its pointless. I think there is a mod if I recall called better sleeping you can set specific times and actually make players sleep for a few in game hours. I could be mistaken on the name tho.
